ocs_ci/ocs/resources/storageconsumer.py

```diff
@@ -86,53 +87,36 @@ def stop_heartbeat(self):
         """
         Suspend status reporter cron job.
         """
-        self._switch_consumer_cluster()
-        patch_param = '{"spec": {"suspend": true}}'
-        self.heartbeat_cronjob.ocp.patch(
-            resource_name=self.heartbeat_cronjob.name, params=patch_param
-        )
-        self._switch_provider_cluster()
+        with config.RunWithConfigContext(self.consumer_context):
```

Comment:
Do we hold some specific consumer context in `self.consumer_context`?
Or why not using new context manager `RunWithFirstConsumerConfigContext` here and other occurrences below?

Reply:
It uses `consumer_context` defined for the instance constructor of the `StorageConsumer` class. It doesn't have to be the first consumer in all cases.